test/mocha/collections_thenables.js

```diff
@@ -197,47 +197,22 @@ describe("Using collection methods with thenables", function() {
 });
 
 describe("Using collection methods with thenables", function() {
-    var name = "join";
-    testFulfillSync(name, function(promise) {
-        return promise.then(function(v){
-            assert(v[0][0].value === 1);
-            assert(v[0][1].value === 2);
-            assert(v[0][2].value === 3);
-            assert(v[1] === 1);
-            assert(v[2] === 2);
-            assert(v[3] === 3);
-        });
-    }, 1, 2, 3);
-    testFulfillAsync(name, function(promise) {
-        return promise.then(function(v){
-            assert(v[0][0].value === 1);
-            assert(v[0][1].value === 2);
-            assert(v[0][2].value === 3);
-            assert(v[1] === 1);
-            assert(v[2] === 2);
-            assert(v[3] === 3);
-        });
-    }, 1, 2, 3);
-    testRejectSync(name, function(promise) {
```

Comment:
The tests were testing `join` in the old syntax. I added new tests for `then`ables with the new syntax. I'll add these tests again with the new syntax.